Source/JavaScriptCore/runtime/CodeCache.cpp

```diff
@@ -67,13 +67,28 @@ static void generateUnlinkedCodeBlockForFunctions(VM& vm, UnlinkedCodeBlock* unl
 
     // FIXME: We should also generate CodeBlocks for CodeForConstruct
     // https://bugs.webkit.org/show_bug.cgi?id=193823
-    for (unsigned i = 0; i < unlinkedCodeBlock->numberOfFunctionDecls(); i++)
-        generate(unlinkedCodeBlock->functionDecl(i), CodeForCall);
-    for (unsigned i = 0; i < unlinkedCodeBlock->numberOfFunctionExprs(); i++)
-        generate(unlinkedCodeBlock->functionExpr(i), CodeForCall);
+    //
+    // NOTE: We changed this in Bun. We check if the function is a constructor
+    // and if it is, we generate a CodeForConstruct block.
+    for (unsigned i = 0; i < unlinkedCodeBlock->numberOfFunctionDecls(); i++) {
```

Comment:
do these need to generate in this order? if not we should change it to one loop `for (unsigned i = 0; i < max(numberOfFunctionDecls(), numberOfFunctionExprs(), i ++`